webapp/src/components/conference/conference.tsx

```diff
@@ -0,0 +1,323 @@
+        };
+    }
+
+    getViewportWidth(): number {
+        return Math.max(document.documentElement.clientWidth || 0, window?.innerWidth || 0) - BORDER_SIZE;
+    }
+
+    getViewportHeight(): number {
+        return Math.max(document.documentElement.clientHeight || 0, window?.innerHeight || 0) - BORDER_SIZE;
+    }
+
+    initJitsi = (post: Post) => {
+        const vw = this.getViewportWidth();
+        const vh = this.getViewportHeight();
+
+        const url = new URL(post.props.meeting_link);
+
+        const noSSL = url.protocol === 'http:';
+
+        const domain = url.hostname;
+        const options = {
+            roomName: post.props.meeting_id,
+            width: this.state.minimized ? MINIMIZED_WIDTH : vw,
+            height: this.state.minimized ? MINIMIZED_HEIGHT : vh,
+            jwt: this.props.jwt,
+            noSSL,
+            parentNode: document.querySelector('#jitsiMeet'),
+            onload: () => {
+                this.setState({loading: false});
+                this.resizeIframe();
+            }
+        };
+        this.api = new (window as any).JitsiMeetExternalAPI(domain, options);
+        this.api.on('videoConferenceJoined', () => {
+            if (this.state.minimized) {
+                this.minimize();
+            }
+        });
+        this.api.on('readyToClose', () => {
+            this.close();
+        });
+        this.api.on('tileViewChanged', (event: {enabled: boolean}) => {
+            if (!this.state.minimized) {
+                this.setState({wasTileView: event.enabled});
+            }
+            this.setState({isTileView: event.enabled});
+        });
+        this.api.on('filmstripDisplayChanged', (event: {visible: boolean}) => {
+            if (!this.state.minimized) {
+                this.setState({wasFilmStrip: event.visible});
+            }
+            this.setState({isFilmStrip: event.visible});
+        });
+        this.api.executeCommand('subject', post.props.meeting_topic);
+    }
+
+    resizeIframe = () => {
+        if (this.api && this.props.post) {
+            const vw = this.getViewportWidth();
+            const vh = this.getViewportHeight();
+            const iframe = this.api.getIFrame();
+            iframe.style.width = this.state.minimized ? MINIMIZED_WIDTH + 'px' : vw + 'px';
+            iframe.style.height = this.state.minimized ? MINIMIZED_HEIGHT + 'px' : vh + 'px';
+        }
+    }
+
+    componentDidMount() {
+        window.addEventListener('resize', this.resizeIframe);
+        if (this.props.post) {
+            this.initJitsi(this.props.post);
+        }
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.resizeIframe);
+        if (this.api) {
+            this.api.dispose();
+        }
+    }
+
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        if (this.props.post) {
+            if (prevProps.post !== this.props.post) {
+                this.initJitsi(this.props.post);
```

Comment:
Since we are creating a new api in initJitsi, should we dispose of the previous one here?

Reply:
Yes, I guess so